src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -580,9 +583,14 @@ def _search_role_assignments(cli_ctx, assignments_client, definitions_client,
     # "atScope()" and "principalId eq '{value}'" query cannot be used together (API limitation).
     # always use "scope" if provided, so we can get assignments beyond subscription e.g. management groups
     if scope:
-        f = 'atScope()'  # atScope() excludes role assignments at subscopes
+        filters = []
+        if at_scope:
+            filters.append('atScope()')  # atScope() excludes role assignments at subscopes
+        if assignee_object_id and not include_groups and not at_scope:
+            filters.append("principalId eq '{}'".format(assignee_object_id))
```

Comment:
Now with the ability to turn off `atScope()`, `principalId eq '{}'` filter can be used with `scope`, fixing https://github.com/Azure/azure-cli/issues/14302#issuecomment-2235937993.